test/request_source/request_source_plugin_test.cc

```diff
@@ -337,5 +394,75 @@ TEST_F(InLineRequestSourcePluginTest,
   EXPECT_EQ(header2->getPathValue(), "/b");
   EXPECT_EQ(header3->getPathValue(), "/a");
 }
+
+TEST_F(
+    InLineRequestSourcePluginTest,
+    CreateRequestSourcePluginMultipleTimesWithDifferentConfigsCreatesDifferentWorkingRequestsSources) {
+  Envoy::MessageUtil util;
+  nighthawk::client::RequestOptionsList options_list_ab;
+  util.loadFromFile(/*file to load*/ Nighthawk::TestEnvironment::runfilesPath(
+                        "test/request_source/test_data/test-config-ab.yaml"),
+                    /*out parameter*/ options_list_ab,
+                    /*validation visitor*/ Envoy::ProtobufMessage::getStrictValidationVisitor(),
+                    /*Api*/ *api_,
+                    /*use api boosting*/ true);
+  nighthawk::client::RequestOptionsList options_list_c;
+  util.loadFromFile(/*file to load*/ Nighthawk::TestEnvironment::runfilesPath(
+                        "test/request_source/test_data/test-config-c.yaml"),
+                    /*out parameter*/ options_list_c,
+                    /*validation visitor*/ Envoy::ProtobufMessage::getStrictValidationVisitor(),
+                    /*Api*/ *api_,
+                    /*use api boosting*/ true);
+
+  nighthawk::request_source::InLineOptionsListRequestSourceConfig config_ab =
+      MakeInLinePluginConfig(options_list_ab, /*num_requests*/ 2);
+  nighthawk::request_source::InLineOptionsListRequestSourceConfig config_c =
+      MakeInLinePluginConfig(options_list_c, /*num_requests*/ 2);
+
+  Envoy::ProtobufWkt::Any config_ab_any;
+  config_ab_any.PackFrom(config_ab);
+  Envoy::ProtobufWkt::Any config_c_any;
+  config_c_any.PackFrom(config_c);
+
+  auto& config_factory =
+      Envoy::Config::Utility::getAndCheckFactoryByName<RequestSourcePluginConfigFactory>(
+          "nighthawk.in-line-options-list-request-source-plugin");
+  auto header_ab = Envoy::Http::RequestHeaderMapImpl::create();
```

Comment:
Can we use the actual std::unique_ptr<TerminationPredicate> type here and below?